### PR TITLE
Add unit tests for `build_script_function`

### PR DESCRIPTION
**Pull request by makslevental**
Add unit tests for `build_script_function`.

NB: this PR is stacked on top of https://github.com/llvm/torch-mlir/pull/740, so only the most recent commit is the actual diff.

### CODE REVIEW

File: python/test/eager_mode/normalize_args_kwargs.py [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+
+import torch
+
+from framework import register_test, run_tests
+from torch_mlir.eager_mode.torch_mlir_dispatch import normalize_args_kwargs
+
+
+# CHECK: PASS - should_normalize
+@register_test
```
Comment (silvasean) [inline python/test/eager_mode/normalize_args_kwargs.py:16]: I don't think you need to "register" anything. Just running it immediately is fine. The requrements of the e2e testing framework are a bit different since we need to drive it from the command line, filter tests, etc. But in this case we already have lit which can do that at .py-file granularity which seems to be enough.

**Review by silvasean — CHANGES_REQUESTED** [review]

File: python/torch_mlir/eager_mode/torch_mlir_dispatch.py [hunk 2]
```diff
@@ -125,7 +126,11 @@ def build_script_function(
     else:
         graph.registerOutput(node.output())
 
-    fn_name = str(node).strip()
+    # The first alternation handles non-alpha characters at the beginning.
+    # MLIR doesn't support such symbol names unless quoted, hence safer to remove.
+    # The second alternation handles non-word characters in the rest of the name (eg =, :, etc).
+    # LLVM doesn't support such symbol names.
+    fn_name = re.sub(r'^[^a-zA-Z]+|[\W_]+', '', str(node))
```
Comment (silvasean) [inline python/torch_mlir/eager_mode/torch_mlir_dispatch.py:133]: I'm not sure I understand the requirements here. In what way is this name load-bearing? Can we just call it "f"? If not, then I think we also need to mangle the tensor shapes we specialized for as well.

Comment (makslevental) [reply python/torch_mlir/eager_mode/torch_mlir_dispatch.py:133]: it's load-bearing because we use this name to pull the exported function out of the module here

https://github.com/llvm/torch-mlir/blob/9ec0683e928bdd1acea9aeb025ed542332fab902/python/torch_mlir/eager_mode/torch_mlir_dispatch.py#L260

>op_mlir_backend_callable = getattr(loaded_module, script_fun.name)

right now there's only ever one exported function in that module but in the future there could be others (in the case that we decide to incrementally compile and inline or not).

>If not, then I think we also need to mangle the tensor shapes we specialized for as well.

good idea!

Comment (silvasean) [reply python/torch_mlir/eager_mode/torch_mlir_dispatch.py:133]: Looking at the code, it seems like it would still continue to work if we just called the function "f" every time. Right? That seems simplest to do so I think we should do it.

File: python/torch_mlir/eager_mode/torch_mlir_dispatch.py [hunk 3]
```diff
@@ -6,6 +6,7 @@
 from typing import Any, Callable, Tuple, Union
 from typing import List, Dict
 
+import re
```
Comment (silvasean) [inline python/torch_mlir/eager_mode/torch_mlir_dispatch.py:9]: nit: can remoe the import

**Review by silvasean — APPROVED** [review]
